Shared/Services/Backups/BackupsService.swift

```diff
@@ -486,6 +486,130 @@ class BackupsService: ObservableObject {
             })
 
     }
+    
+    func downloadFolderBackup(device: Device, downloadAt: URL, folderId: String) async throws {
+        self.deviceDownloading = device
+        logger.info("Preparint  folder backup for download")
+      
+        DispatchQueue.main.sync {
+            self.backupDownloadStatus = .InProgress
+            self.backupDownloadedItems = 0
+        }
+        guard let deviceBucketId = device.bucket else {
+            logger.error("Bucket id is nil")
+            throw BackupError.bucketIdIsNil
+        }
+
+
+        logger.info("Setting connection to XPCBackupService...")
+        
+        let xpcBackupService = try await getXPCBackupServiceProtocol(onConnectionIssue: {
+            self.logger.error("❌ XPCBackupService connection interrupted")
+        })
+        
+        logger.info("✅ Connection to XPCBackupService stablished")
+        let configLoader = ConfigLoader()
+        let networkAuth = configLoader.getNetworkAuth()
+        
+        guard let URLAsString = downloadAt.absoluteString.replacingOccurrences(of: "file://", with: "").removingPercentEncoding else {
+            throw BackupError.invalidDownloadURL
+        }
+
+        guard let networkAuthUnwrapped = networkAuth else {
+            throw BackupError.missingNetworkAuth
+        }
+        
+        backupDownloadProgressTimer?.cancel()
+        xpcBackupService.downloadFolderBackup(
+            downloadAt: URLAsString,
+            networkAuth: networkAuthUnwrapped,
+            folderId:folderId,
+            bucketId: deviceBucketId,
+            with: {result, error in
+                if error == nil {
+                    DispatchQueue.main.async {
+                        self.backupDownloadStatus = .Done
```

Comment:
watch out, this status is for when you are downloading an entire backup, I think we should not use this for this operation